docs/source/exporters/onnx/usage_guides/contribute.mdx

```diff
@@ -0,0 +1,225 @@
+<!--Copyright 2022 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+## Adding support for an unsupported architecture
+
+If you wish to export a model whose architecture is not already supported by the library, there are three main steps to follow:
+
+1. Implement a custom ONNX configuration.
+2. Register the ONNX configuration in the [`~optimum.exporters.TasksManager`]
+2. Export the model to ONNX.
+3. Validate the outputs of the original and exported models.
+
+In this section, we'll look at how BERT was implemented to show what's involved with each step.
+
+### Implementing a custom ONNX configuration
+
+Let's start with the ONNX configuration object. We provide a 3-level (class hierarchy)[package_reference/configuration],
+and to add support for a model, inheriting from the right middle-end class will be the way to go most of the time. You might have to
+implement a middle-end class yourself if you are adding an architecture handling a modality and/or case never seen before.
+
+<Tip>
+
+A good way to implement a custom ONNX configuration is to look at the existing configuration implementation in the
+`optimum/exporters/onnx/model_configs.py` file.
+
+Also, if the architecture you are trying to add is (very) similar to an architecture that is already supported
+(for instance adding support for ALBERT when BERT is already supported), trying to simply inheriting from this class
+might work.
+
+</Tip>
+
+
+When inheriting from a middle-end class, look for the one handling the same modality / category of model as the one you
+are trying to support.
+
+#### Example: Adding support for BERT
+
+Since BERT is an encoder-based model for text, its configuration inherits from the middle-end class [`~optimum.exporters.onnx.config.TextEncoderOnnxConfig`].
+In `optimum/exporters/onnx/model_configs.py`:
+
+```python
+# This class is actually in optimum/exporters/onnx/config.py
+class TextEncoderOnnxConfig(OnnxConfig):
+    DUMMY_INPUT_GENERATOR_CLASSES = (DummyTextInputGenerator,)
```

Comment:
Idea: add brief comments to explain what each of these attributes does for those too lazy to read the main text :)?